swan-lake/featured-scenarios/work-with-data-using-queries-in-ballerina.md

```diff
@@ -426,4 +426,266 @@ Countries with highest deaths:[["India",980976],["USA",880976],["Afghanistan",73
+    return countriesWithRecovered;
+}
+
+public function printErroneousData(table<CovidEntry> dataTable) {
+    string[] countries = from CovidEntry entry in dataTable
+        let decimal sum = entry.recovered + entry.deaths + entry.active
+        where entry.cases != sum
+        select entry.country;
+
+    if countries.length() > 0 {
+        io:println("Found erroneous entries for countries: ", countries);
+    }
+}
+
+public function groupDeathsByContinent(table<CovidEntry> dataTable) returns record {|string continent; decimal deaths;|}[] {
+    return from var {continent, deaths} in dataTable
+        group by continent
+        select {continent, deaths: sum(deaths)};
+}
+
+public function main() {
+    string[] countries = filterCountriesByCases(covidTable, 10000000);
+    io:println("Countries with more than 10 million cases: ", countries);
+
+    [string, decimal][] countriesWithDeaths = findCountriesByHighestNoOfDeaths(covidTable, 3);
+    io:println("Countries with highest deaths: ", countriesWithDeaths);
+
+    string[] c = ["USA", "India", "Afghanistan"];
+    [string, decimal][] countriesWithRecovered = findRecoveredPatientsOfCountries(covidTable, c);
+    io:println("Countries with number of Recovered patients: ", countriesWithRecovered);
+
+    printErroneousData(covidTable);
+
+    record {|string continent; decimal deaths;|}[] deathsByContinent = groupDeathsByContinent(covidTable);
+    io:println("Total number of deaths by continent: ", deathsByContinent);
+}
+```
+
+### Run the package for grouping
+
+In the terminal, navigate to the `query_expressions` directory, and execute the command below to run the service package.
+
+```bash
+$ bal run
+```
+
+You view the output below.
+
+```bash
+Compiling source
+	    .../query_expressions:0.1.0
+
+Running executable
+
+Compiling source
+	kavindu/query_expressions:0.1.0
+
+Running executable
+
+Countries with more than 10 million cases: ["USA","India"]
+Countries with highest deaths:[["India",980976],["USA",880976],["Afghanistan",7386]]
+Countries with number of Recovered patients:[["Afghanistan",146084],["USA",43892277],["India",33892279]]
+Found erroneous entries for countries: ["Sri Lanka","India"]
+Total number of deaths by continent: [{"continent":"Asia","deaths":993605},{"continent":"North America","deaths":880976}]
+```
+
+## Get the total COVID-19 deaths
+
+### Create the `getTotalNumberOfDeaths` function
+
+To define a function to get total number of deaths, add the code below to the `main.bal` file.
+
+```ballerina
+public function getTotalNumberOfDeaths(table<CovidEntry> dataTable) returns decimal {
+    return from var {deaths} in dataTable
+            collect sum(deaths);
+}
+```
+
+In this code,
+- The `collect` clause collects all input frames into one group.
+- Since `deaths` becomes a sequence value, you can call the `sum` function in order to get the total number of deaths.
+
+### Update the main function for collecting
```

Comment:
"collect" is just a keyword I made up. I don't think "collecting" is the normal term for this. I suggest "aggregating".

Pedagogically I think it's better to start with a section on "Aggregating" that introduces "collect", and use aggregation functions (like `sum). Then have a section on "Grouping" that introduces "group by".